ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestConfig.java

```diff
@@ -90,31 +96,28 @@
 
   public static final String AUTHENTICATION_ROLES_CONFIG = "authentication.roles";
   public static final String AUTHENTICATION_ROLES_DOC = "Valid roles to authenticate against.";
-  public static final List<String> AUTHENTICATION_ROLES_DEFAULT =
-      Collections.unmodifiableList(Arrays.asList("*"));
+  public static final List<String> AUTHENTICATION_ROLES_DEFAULT = Collections.singletonList("*");
 
-  public static final String SSL_KEYSTORE_LOCATION_CONFIG = "ssl.keystore.location";
-  protected static final String SSL_KEYSTORE_LOCATION_DOC =
-      "Location of the keystore file to use for SSL. This is required for HTTPS.";
   protected static final String SSL_KEYSTORE_LOCATION_DEFAULT = "";
-  public static final String SSL_KEYSTORE_PASSWORD_CONFIG = "ssl.keystore.password";
-  protected static final String SSL_KEYSTORE_PASSWORD_DOC =
-      "The store password for the keystore file.";
   protected static final String SSL_KEYSTORE_PASSWORD_DEFAULT = "";
 
-  public static final String SSL_TRUSTSTORE_LOCATION_CONFIG = "ssl.truststore.location";
-  protected static final String SSL_TRUSTSTORE_LOCATION_DOC =
-      "Location of the trust store. Required only to authenticate HTTPS clients.";
   protected static final String SSL_TRUSTSTORE_LOCATION_DEFAULT = "";
-  public static final String SSL_TRUSTSTORE_PASSWORD_CONFIG = "ssl.truststore.password";
-  protected static final String SSL_TRUSTSTORE_PASSWORD_DOC =
-      "The store password for the trust store file.";
   protected static final String SSL_TRUSTSTORE_PASSWORD_DEFAULT = "";
 
-  public static final String SSL_CLIENT_AUTH_CONFIG = "ssl.client.auth";
-  protected static final String SSL_CLIENT_AUTH_DOC =
-      "Whether or not to require the https client to authenticate via the server's trust store.";
-  protected static final boolean SSL_CLIENT_AUTH_DEFAULT = false;
+  public static final String SSL_CLIENT_AUTHENTICATION_CONFIG = "ssl.client.authentication";
```

Comment:
This PR contains breaking changes, such as this config rename from `ssl.client.auth` to `ssl.client.authentication`. Can you add a `BREAKING CHANGE:` description into the PR description and commit message (when merging this PR) with details?

I think we should also add a check for the old config (`ssl.client.auth`) and fail startup if the old config is specified, in order to alert users to the breaking change, since the current behavior of silently ignoring the old config and using the default value of the new config does not seem acceptable.

Reply:
ack

Reply:
What I've done is added code so we support the old deprecated client auth and warn in the logs if it's used. Basically the same check that rest-utils does here https://github.com/confluentinc/rest-utils/blob/master/core/src/main/java/io/confluent/rest/ApplicationServer.java#L206

So it shouldn't be.a breaking change.

Reply:
Sounds good, though we should update the docs to introduce the new config and say the old one is deprecated.